src/NHibernate.Test/NHSpecificTest/NH3932/Fixture.cs

```diff
@@ -0,0 +1,205 @@
+﻿using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.NH3932
+{
+	public abstract class Fixture : BugTestCase
+	{
+		private IParent storedParent;
+		protected abstract bool CareAboutOrder { get; }
+
+		[Test]
+		public void ShouldKeepDirtyCollectionDirtyAfterMergingClone()
+		{
+			using (var s = OpenSession())
+			{
+				using (s.BeginTransaction())
+				{
+					s.Lock(storedParent, LockMode.None);
+					storedParent.ClearChildren();
+					Assert.IsTrue(s.IsDirty());
+
+					var loadedClone = storedParent.Clone();
+					s.Merge(loadedClone);
+					Assert.IsTrue(s.IsDirty());
+				}
+			}
+		}
+
+		[Test]
+		public void ShouldCareAboutOrder_MergeObjectNotInSession()
+		{
+			var parent = CreateParent(2);
+			using (var s = OpenSession())
+			{
+				using (var tx = s.BeginTransaction())
+				{
+					s.Save(parent);
+					tx.Commit();
+				}
+			}
+			var parentClone = parent.Clone();
+			parentClone.ReverseChildren();
+			using (var s = OpenSession())
+			{
+				s.SessionFactory.Statistics.Clear();
+				using (var tx = s.BeginTransaction())
+				{
+					s.Merge(parentClone);
+					tx.Commit();
+				}
+				Assert.That(s.SessionFactory.Statistics.EntityUpdateCount, CareAboutOrder ? Is.EqualTo(1) : Is.EqualTo(0));
+			}
+		}
+
+		[Test]
+		public void ShouldCareAboutOrder_MergeObjectInSession()
+		{
+			var parent = CreateParent(2);
+			using (var s = OpenSession())
+			{
+				using (var tx = s.BeginTransaction())
+				{
+					s.Save(parent);
+					tx.Commit();
+				}
+			}
+			var parentClone = parent.Clone();
+			parentClone.ReverseChildren();
+			using (var s = OpenSession())
+			{
+				s.SessionFactory.Statistics.Clear();
+				using (var tx = s.BeginTransaction())
+				{
+					s.Lock(parent, LockMode.None);
+					s.Merge(parentClone);
+					tx.Commit();
+				}
+				Assert.That(s.SessionFactory.Statistics.EntityUpdateCount, CareAboutOrder ? Is.EqualTo(1) : Is.EqualTo(0));
+			}
+		}
+
+		[Test]
+		public void MergeCleanCloneShouldNotResultInUpdate()
+		{
+			var parentClone = storedParent.Clone();
+			using (var s = OpenSession())
+			{
+				s.SessionFactory.Statistics.Clear();
+				using (var tx = s.BeginTransaction())
+				{
+					s.Merge(parentClone);
+					tx.Commit();
+				}
+				Assert.That(s.SessionFactory.Statistics.EntityUpdateCount, Is.EqualTo(0));
+			}
+		}
+
+		[Test]
+		public void MergeCleanCloneShouldNotMakeSessionDirty()
+		{
+			var parentClone = storedParent.Clone();
+			using (var s = OpenSession())
+			{
+				s.SessionFactory.Statistics.Clear();
+				using (var tx = s.BeginTransaction())
+				{
+					s.Merge(parentClone);
+					tx.Commit();
+				}
+				Assert.That(s.IsDirty(), Is.False);
```

Comment:
Yes, my fault! Was added as an "extra" test next to the one counting entityupdatecount.... So this one was probably never red... :(.

Fixed.